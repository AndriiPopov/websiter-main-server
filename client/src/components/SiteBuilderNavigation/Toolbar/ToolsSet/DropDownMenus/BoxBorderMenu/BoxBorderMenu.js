import React from 'react';
import { connect } from 'react-redux';

import * as actions from '../../../../../../store/actions/index';
import ValueInput from '../../../../../UI/ValueInput/ValueInput';
import RangeInput from '../../../../../UI/RangeInput/RangeInput';
import Checkbox from '../../../../../UI/Checkbox/Checkbox';
import BigColorPicker from '../../../../../UI/Buttons/BigColorPicker/BigColorPicker';

export const BoxBorderMenu = props => {
    return (
        <table>
            <tbody>
                <tr>
                    <td>On</td>
                    <td colSpan="2">
                        <Checkbox
                            onChange={(value) => props.changeElementPropertyValue(value, 'borderOn')} />
                    </td>
                </tr>
                <tr>
                    <td>Border color</td>
                    <td colSpan="2">
                        <BigColorPicker
                            color="rgba(0, 0, 0, 1)"
                            boxesValues={props.boxesValues}
                            changedStyleValues={props.changedStyleValues}
                            saveToHistory={() => props.saveToHistory()}
                            changed={color => props.changed(color, 'borderColor')} />
                    </td>
                </tr>
                <tr>
                    <td>Size</td>
                    <td>
                        <RangeInput
                            changed={newValue => props.changed(newValue, 'borderSize')}
                            startValue={props.boxesValues.borderSize}
                            saveToHistory={() => props.saveToHistory()}
                            step="1"
                            min="0"
                            max="20" />
                    </td>
                    <td>
                        <ValueInput
                            startValue={props.boxesValues.borderSize}
                            changed={value => props.changed(value, 'borderSize')} />
                    </td>
                </tr>
                {/* <tr>
                    <td>Radius</td>
                    <td><RangeInput
                        changed="changeElementPropertyValue"
                        valueKey="borderRadius"
                        max="20" /></td>
                    <td><ValueInput 
                        valueKey="borderRadius"
                        changed="changeElementPropertyValue" /></td>
                </tr> */}
                <tr>
                    <td>H offset</td>
                    <td>
                        <RangeInput
                            changed={newValue => props.changed(newValue, 'borderHOffset')}
                            startValue={props.boxesValues.borderHOffset}
                            saveToHistory={() => props.saveToHistory()}
                            valueKey=""
                            min="-20"
                            max="20" />
                        </td>
                    <td>
                        <ValueInput 
                            startValue={props.boxesValues.borderHOffset}
                            changed={value => props.changed(value, 'borderHOffset')} />
                        </td>
                </tr>
                <tr>
                    <td>V offset</td>
                    <td>
                        <RangeInput
                            changed={newValue => props.changed(newValue, 'borderVOffset')}
                            startValue={props.boxesValues.borderVOffset}
                            saveToHistory={() => props.saveToHistory()}
                            min="-20"
                            max="20" />
                        </td>
                    <td>
                        <ValueInput 
                            startValue={props.boxesValues.borderVOffset}
                            changed={value => props.changed(value, 'borderVOffset')} />
                    </td>
                </tr>
                <tr>
                    <td>Blur</td>
                    <td>
                        <RangeInput
                            changed={newValue => props.changed(newValue, 'borderBlur')}
                            saveToHistory={() => props.saveToHistory()}
                            startValue={props.boxesValues.borderBlur}
                            max="20" />
                        </td>
                    <td>
                        <ValueInput 
                            startValue={props.boxesValues.borderBlur}
                            changed={value => props.changed(value, 'borderBlur')} />
                    </td>
                </tr>
                <tr>
                    <td>Inset</td>
                    <td><Checkbox 
                        changed="changeElementPropertyValue"
                        valueKey="borderInset" /></td>
                    <td></td>
                </tr>
            </tbody>
        </table>
    );
};

const mapStateToProps = state => {
    return {
        boxesValues: state.builder.present.chosenBoxesValues,
        changedStyleValues: state.builder.present.changedStyleValues
    }
}

const mapDispatchToProps = dispatch => {
    return {
        saveToHistory: () => { dispatch(actions.saveToHistory()) },
        changed: (value, valueKey) => dispatch(actions.changeElementPropertyValue(value, valueKey, false, true))
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(BoxBorderMenu);
        