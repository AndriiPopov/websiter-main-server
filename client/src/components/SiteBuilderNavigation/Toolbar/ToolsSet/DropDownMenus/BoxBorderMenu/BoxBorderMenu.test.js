import React from 'react';
import {shallow, configure} from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import { BoxBorderMenu } from './BoxBorderMenu';

configure({adapter: new Adapter()});

describe('BoxBorderMenu', () => {
    it('should be defined', () => {
        expect(BoxBorderMenu).toBeDefined();
    });

    it('it should render a dropdown align menu', () => {
        const tree = shallow(
            <BoxBorderMenu
                alignMode="parent" />
        );
        expect(toJson(tree)).toMatchSnapshot();
    });
});