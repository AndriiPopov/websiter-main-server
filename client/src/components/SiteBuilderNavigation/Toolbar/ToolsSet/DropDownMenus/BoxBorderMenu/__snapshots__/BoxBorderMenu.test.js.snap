// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxBorderMenu it should render a dropdown align menu 1`] = `
<table>
  <tbody>
    <tr>
      <td>
        Border color
      </td>
      <td>
        <Connect(BigColorPicker)
          changed="changeElementPropertyValue"
          color="rgba(0, 0, 0, 1)"
          valueKey="borderColor"
        />
      </td>
      <td />
    </tr>
    <tr>
      <td>
        Size
      </td>
      <td>
        <Connect(RangeInput)
          changed="changeElementPropertyValue"
          max="20"
          min="0"
          step="1"
          valueKey="borderSize"
        />
      </td>
      <td>
        <Connect(ValueInput)
          changed="changeElementPropertyValue"
          valueKey="borderSize"
        />
      </td>
    </tr>
    <tr>
      <td>
        Radius
      </td>
      <td>
        <Connect(RangeInput)
          changed="changeElementPropertyValue"
          max="20"
          valueKey="borderRadius"
        />
      </td>
      <td>
        <Connect(ValueInput)
          changed="changeElementPropertyValue"
          valueKey="borderRadius"
        />
      </td>
    </tr>
    <tr>
      <td>
        H offset
      </td>
      <td>
        <Connect(RangeInput)
          changed="changeElementPropertyValue"
          max="20"
          min="-20"
          valueKey="borderHOffset"
        />
      </td>
      <td>
        <Connect(ValueInput)
          changed="changeElementPropertyValue"
          valueKey="borderHOffset"
        />
      </td>
    </tr>
    <tr>
      <td>
        V offset
      </td>
      <td>
        <Connect(RangeInput)
          changed="changeElementPropertyValue"
          max="20"
          min="-20"
          valueKey="borderVOffset"
        />
      </td>
      <td>
        <Connect(ValueInput)
          changed="changeElementPropertyValue"
          valueKey="borderVOffset"
        />
      </td>
    </tr>
    <tr>
      <td>
        Blur
      </td>
      <td>
        <Connect(RangeInput)
          changed="changeElementPropertyValue"
          max="20"
          valueKey="borderBlur"
        />
      </td>
      <td>
        <Connect(ValueInput)
          changed="changeElementPropertyValue"
          valueKey="borderBlur"
        />
      </td>
    </tr>
    <tr>
      <td>
        Inset
      </td>
      <td>
        <Connect(Checkbox)
          changed="changeElementPropertyValue"
          valueKey="borderInset"
        />
      </td>
      <td />
    </tr>
  </tbody>
</table>
`;
